homeassistant/components/ipp/config_flow.py

```diff
@@ -0,0 +1,160 @@
+
+        # pylint: disable=no-member # https://github.com/PyCQA/pylint/issues/3167
+        self.context.update({"title_placeholders": {"name": name}})
+
+        self.discovery_info.update(
+            {
+                CONF_HOST: host,
+                CONF_PORT: port,
+                CONF_SSL: tls,
+                CONF_VERIFY_SSL: False,
+                CONF_BASE_PATH: "/"
+                + discovery_info["properties"].get("rp", "ipp/print"),
+                CONF_NAME: name,
+                CONF_UUID: discovery_info["properties"].get("UUID"),
+            }
+        )
+
+        if self.discovery_info[CONF_UUID] is None:
+            try:
+                info = await validate_input(self.hass, self.discovery_info)
+            except IPPConnectionUpgradeRequired:
+                return self.async_abort(reason="connection_upgrade")
+            except IPPConnectionError:
+                return self.async_abort(reason="connection_error")
+            self.discovery_info[CONF_UUID] = info[CONF_UUID]
+
+        await self.async_set_unique_id(self.discovery_info[CONF_UUID])
+        self._abort_if_unique_id_configured(
+            updates={CONF_HOST: self.discovery_info[CONF_HOST]}
+        )
+
+        return await self.async_step_zeroconf_confirm()
+
+    async def async_step_zeroconf_confirm(
+        self, user_input: ConfigType = None
+    ) -> Dict[str, Any]:
+        """Handle a flow initiated by zeroconf."""
+        if user_input is None:
+            return self._show_confirm_dialog()
+
+        try:
+            info = await validate_input(self.hass, self.discovery_info)
+        except IPPConnectionUpgradeRequired:
+            return self.async_abort(reason="connection_upgrade")
+        except IPPConnectionError:
+            return self.async_abort(reason="connection_error")
+        self.discovery_info[CONF_UUID] = info[CONF_UUID]
+
+        await self.async_set_unique_id(self.discovery_info[CONF_UUID])
+        self._abort_if_unique_id_configured(
+            updates={CONF_HOST: self.discovery_info[CONF_HOST]}
+        )
+
+        return self.async_create_entry(
+            title=self.discovery_info[CONF_NAME], data=self.discovery_info,
+        )
+
+    def _show_setup_form(self, errors: Optional[Dict] = None) -> Dict[str, Any]:
+        """Show the setup form to the user."""
+        return self.async_show_form(
+            step_id="user",
+            data_schema=vol.Schema(
+                {
+                    vol.Required(CONF_HOST): str,
+                    vol.Required(CONF_PORT, default=631): int,
+                    vol.Required(CONF_BASE_PATH, default="/ipp/print"): str,
+                    vol.Required(CONF_SSL, default=False): bool,
+                    vol.Required(CONF_VERIFY_SSL, default=False): bool,
+                }
+            ),
+            errors=errors or {},
+        )
+
+    def _show_confirm_dialog(self, errors: Optional[Dict] = None) -> Dict[str, Any]:
```

Comment:
I would just inline this function, only used once.

Reply:
done